meson.build

```diff
@@ -107,6 +108,8 @@ libinput = dependency('libinput', version: '>=1.14.0')
 xkbcommon = dependency('xkbcommon')
 udev = dependency('libudev')
 pixman = dependency('pixman-1')
+uuid = dependency('uuid', required: false)
+uuid_create = cc.has_function('uuid_create')
```

Comment:
I think we should have a Meson feature option for xdg-foreign, just like we have for all other features. Package maintainers don't want auto-detection of features, they want hard failures when a dep is missing.

Reply:
I have added this, let me know what you think of the approach.